examples/08_interpolation_join.py

```diff
@@ -0,0 +1,179 @@
+# We see that in this case the interpolation join works well for the temperature, but not precipitation nor snow.
+# So we will only add the temperature to our flights table.
+
+aux_table = aux_table.drop(["PRCP", "SNOW"], axis=1)
+
+######################################################################
+# Loading the flights table
+# -------------------------
+# We load the flights table and join it to the airports table using the flights’ "Origin" which refers to the departure airport’s IATA code.
+# We use only a subset to speed up the example.
+
+flights = fetch_figshare("41771418").X[["Year_Month_DayofMonth", "Origin", "ArrDelay"]]
+flights = flights.sample(20_000, random_state=0, ignore_index=True)
+airports = fetch_figshare("41710257").X.loc[
+    :, ["iata", "airport", "state", "lat", "long"]
+]
+flights = flights.merge(airports, left_on="Origin", right_on="iata")
+# printing the first row is more readable than the head() when we have many columns
+flights.iloc[0]
+
+######################################################################
+# Joining the flights and weather data
+# ------------------------------------
+# As before, we initialize our join transformer with the weather table.
+# Then, we use it to transform the flights table -- it adds a "TMAX" column containing the predicted maximum daily temperature.
+#
+
+joiner = InterpolationJoiner(
+    aux_table,
+    main_key=["lat", "long", "Year_Month_DayofMonth"],
+    aux_key=["LATITUDE", "LONGITUDE", "YEAR/MONTH/DAY"],
+)
+join = joiner.fit_transform(flights)
+join.head()
+
+######################################################################
+# Sanity checks
+# -------------
+# This time we do not have a ground truth for the temperatures.
+# We can perform a few basic sanity checks.
+
+state_temperatures = join.groupby("state")["TMAX"].mean().sort_values()
+
+######################################################################
+# states with the lowest average predicted temperatures: Alaska, Montana, North Dakota, Washington, Minnesota
+state_temperatures.head()
+
+######################################################################
+# states with the highest predicted temperatures: Puerto Rico, Virgin Islands, Hawaii, Florida, Louisiana
+state_temperatures.tail()
+
+######################################################################
+# Higher latitudes (farther up north) are colder -- the airports in this dataset are in the United States.
+fig, ax = plt.subplots()
+ax.scatter(join["lat"], join["TMAX"])
+ax.set_xlabel("Latitude (higher is farther north)")
+ax.set_ylabel("TMAX")
+
+######################################################################
+# Winter months are colder than spring -- in the north hemisphere January is colder than April
+#
+
+import seaborn as sns
+
+join["month"] = join["Year_Month_DayofMonth"].dt.strftime("%m %B")
+plt.figure()
+sns.barplot(data=join.sort_values(by="month"), y="month", x="TMAX")
```

Comment:
[We have a rendering issue here](https://output.circle-artifacts.com/output/job/289f832a-cccf-41de-85f2-5a9b8b15bdc2/artifacts/0/doc/auto_examples/08_interpolation_join.html#sphx-glr-auto-examples-08-interpolation-join-py). Can we move the plot a bit to the right so that the y-axis labels are seen?